frontend/src/router/routeHandlers.ts

```diff
@@ -77,14 +78,25 @@ export const beforeEachRouteHandler = async (
         // The RouteError will be emitted to a state.
         const routeError = new FamRouteError(
             RouteErrorName.NOT_AUTHENTICATED_ERROR,
-            "You're not login",
+            "You're not logged in. Please login.",
             { to, from }
         );
         emitRouteToastError(routeError);
         // Back to Landing after emit error.
+        setNavigationPath(to.path);
         return { path: routeItems.landing.path };
     }
 
+    if (AuthService.getters.isLoggedIn() && to.path != routeItems.accessRequest.path && !AuthService.getters.userHasRoles()) {
+        const routeError = new FamRouteError(
+            RouteErrorName.NO_ROLES_ERROR,
+            "You do not have any FAM roles. Please request access.",
+            { to, from }
+        );
+        emitRouteToastError(routeError);
+        return { path: routeItems.accessRequest.path };
```

Comment:
I like the idea to check if user has role or not, but might need some discussion about where we direct them to...